x/orderbook/amount.go

```diff
@@ -0,0 +1,16 @@
+package orderbook
+
+
+func NewAmount(whole int64, fractional int64) Amount {
+	return Amount{
+		Whole: whole,
+		Fractional: fractional,
+	}
+}
+
+func NewAmountp(whole int64, fractional int64) *Amount {
```

Comment:
Why have 2 versions of constructor one with pointer and other without? Is this a golang best practice or how it will get handy using weave?

Reply:
Protobuf uses pointers for embedding structs, this can use nil to represent that data not being present, and then a struct if it is present (wire encoding for all fields is optional).

Since you cannot write:
```go
  foo := &CreateOrderMsg{
    price: &NewAmount(1, 3),
  }
```

we use `NewAmountp` (like `coin.NewCoinp`) as a helper for this.